generators/Exercises/RationalNumbers.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Generators.Input;
+using Generators.Output;
+using Newtonsoft.Json.Linq;
+using Humanizer;
+
+namespace Generators.Exercises
+{
+	public struct RationalNumber
+    {
+		public RationalNumber(int[] n)
+		{
+			this.Numerator = n[0];
+			this.Denominator = n[1];
+		}
+		public int Numerator { get; }
+		public int Denominator { get; }
+    }
+	
+    public class RationalNumbers : GeneratorExercise
+    {
+        protected override string RenderTestMethodBodyAssert(TestMethodBody testMethodBody)
+        {
+            var input = testMethodBody.CanonicalDataCase.Properties["input"] as System.Collections.Generic.Dictionary<string, object>;
+            var operation = testMethodBody.CanonicalDataCase.Properties["property"].ToString();
+            var expected = testMethodBody.CanonicalDataCase.Properties["expected"];
+			var operationName = char.ToUpper(operation[0]) + operation.Substring(1);			
+            string template = "Assert.Equal([expression]);";
+			string expression = "";
+			
+            switch (operation.ToLower())
+            {
+                case "add":
+                case "sub":
+                case "mul":
+                case "div":
+				    {
+                       var r1 = new RationalNumber((int[])input["r1"]);
+                       var r2 = new RationalNumber((int[])input["r2"]);
+                       var e = new RationalNumber((int[])expected);
+                       expression = $"new RationalNumber ({e.Numerator},{e.Denominator}), new RationalNumber({r1.Numerator},{r1.Denominator}).{operationName}(new RationalNumber({r2.Numerator},{r2.Denominator}))";
```

Comment:
There should be a space between the different constructor arguments.

Reply:
@ErikSchierboom OK, I'll do indentation etc. fixes today.

Should the RationalNumbers.csproj to be also added to the Exercises.sln to have 
the Travis CI build succeeding?

FYI: I have temporarily added +,-,*, and / operators overloading to the Example.cs class, e.g.:

    public static RationalNumber operator+ (RationalNumber r1, RationalNumber r2)
    {
        return r1.Add(r2);
    }

with the test code as the following:

    [Fact]
    public void Add_two_positive_rational_numbers()
    {
        Assert.Equal(new RationalNumber(7, 6), new RationalNumber(1, 2).Add(new RationalNumber(2, 3)));
        Assert.Equal(new RationalNumber(7, 6), new RationalNumber(1, 2) + new RationalNumber(2, 3));
    }

Leave it or remove?

Thank you.

Reply:
I don't think you should have to add the project to the solution, but there is also no harm in doing that.

I love the idea of introducing custom operator overloading in this exercise, it really makes a lot of sense here. I do think is that this should bump up the difficulty to 4. Additionally, could you also add a HINTS.md file (see [example](https://github.com/exercism/csharp/blob/24dcec7552b0c417ae22c4c759ba4c1d410d0d71/exercises/accumulate/.meta/hints.md)) providing a link to official C# operator overloading documentation?